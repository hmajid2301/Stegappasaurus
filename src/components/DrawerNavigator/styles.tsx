import {StyleSheet} from 'react-native';

import {primary} from '~/constants/colors';
import {body} from '~/constants/fonts';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },

  header: {
    alignItems: 'center',
    borderBottomColor: primary,
    borderBottomWidth: 2,
    height: 80,
    justifyContent: 'center',
    paddingTop: 15,
  },

  icons: {
    width: 30,
  },

  text: {
    fontFamily: body,
    fontWeight: 'normal',
  },
});

export default styles;
