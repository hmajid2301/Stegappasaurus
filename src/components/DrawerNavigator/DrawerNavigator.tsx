import React from "react";
import { View } from "react-native";
import { DrawerItems, DrawerItemsProps } from "react-navigation";

import { ITheme } from "@types";
import Logo from "~/components/Logo";
import { colors } from "~/modules";
import styles from "./styles";

const DrawerNavigator: React.FunctionComponent<DrawerItemsProps> = props => {
  if (props.screenProps !== undefined) {
    const theme: ITheme = props.screenProps.theme;

    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <View style={[styles.header, { backgroundColor: theme.background }]}>
          <Logo color={theme.color} isDark={theme.isDark} />
        </View>
        <DrawerItems
          activeBackgroundColor={colors.primary}
          activeTintColor={colors.pureWhite}
          {...props}
        />
      </View>
    );
  }
  return <View />;
};

export default DrawerNavigator;
