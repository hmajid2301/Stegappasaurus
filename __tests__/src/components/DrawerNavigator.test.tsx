import { shallow } from "enzyme";
import React from "react";

import DrawerNavigator from "~/components/DrawerNavigator";

describe("DrawerNavigator: Match snapshots", () => {
  test("1", () => {
    const component = shallow(
      // @ts-ignore
      <DrawerNavigator
        activeItemKey="Home"
        drawerPosition="left"
        items={[
          {
            routes: [
              {
                key: "Encoding",
                isTransitioning: false,
                index: 0,
                routes: [],
                routeName: "Encoding"
              },
              {
                key: "Decoding",
                isTransitioning: false,
                index: 0,
                routes: [],
                routeName: "Decoding"
              }
            ],
            index: 0,
            isTransitioning: false,
            key: "Home",
            routeName: "Home"
          }
        ]}
        navigation={
          {
            state: {
              routes: [
                {
                  routes: [
                    {
                      key: "Encoding",
                      isTransitioning: false,
                      index: 0,
                      routes: [],
                      routeName: "Encoding"
                    },
                    {
                      key: "Decoding",
                      isTransitioning: false,
                      index: 0,
                      routes: [],
                      routeName: "Decoding"
                    }
                  ],
                  index: 0,
                  isTransitioning: false,
                  key: "Home",
                  routeName: "Home"
                }
              ],
              index: 0,
              isTransitioning: false,
              isDrawerOpen: false,
              key: "Hello",
              params: { car_id: 123 }
            }
          } as any
        }
        screenProps={{
          theme: {
            background: "#FFF",
            color: "#000",
            isDark: false
          }
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test("2", () => {
    const component = shallow(
      // @ts-ignore
      <DrawerNavigator
        activeItemKey="Home"
        drawerPosition="left"
        items={[
          {
            routes: [
              {
                key: "Encoding",
                isTransitioning: false,
                index: 0,
                routes: [],
                routeName: "Encoding"
              },
              {
                key: "Decoding",
                isTransitioning: false,
                index: 0,
                routes: [],
                routeName: "Decoding"
              }
            ],
            index: 0,
            isTransitioning: false,
            key: "Home",
            routeName: "Home"
          }
        ]}
        navigation={
          {
            state: {
              routes: [
                {
                  routes: [
                    {
                      key: "Encoding",
                      isTransitioning: false,
                      index: 0,
                      routes: [],
                      routeName: "Encoding"
                    },
                    {
                      key: "Decoding",
                      isTransitioning: false,
                      index: 0,
                      routes: [],
                      routeName: "Decoding"
                    }
                  ],
                  index: 0,
                  isTransitioning: false,
                  key: "Home",
                  routeName: "Home"
                }
              ],
              index: 0,
              isTransitioning: false,
              isDrawerOpen: false,
              key: "Hello",
              params: { car_id: 123 }
            }
          } as any
        }
      />
    );
    expect(component).toMatchSnapshot();
  });
});
