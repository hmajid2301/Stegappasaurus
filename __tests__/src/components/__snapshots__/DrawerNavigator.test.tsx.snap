// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerNavigator: Match snapshots 1 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#FFF",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomColor": "#009CFF",
          "borderBottomWidth": 2,
          "height": 80,
          "justifyContent": "center",
          "paddingTop": 15,
        },
        Object {
          "backgroundColor": "#FFF",
        },
      ]
    }
  >
    <Logo
      color="#000"
      isDark={false}
    />
  </View>
  <DrawerNavigatorItems
    activeBackgroundColor="#009CFF"
    activeItemKey="Home"
    activeTintColor="#FFF"
    drawerPosition="left"
    iconContainerStyle={
      Object {
        "width": 30,
      }
    }
    inactiveBackgroundColor="transparent"
    inactiveTintColor="#000"
    items={
      Array [
        Object {
          "index": 0,
          "isTransitioning": false,
          "key": "Home",
          "routeName": "Home",
          "routes": Array [
            Object {
              "index": 0,
              "isTransitioning": false,
              "key": "Encoding",
              "routeName": "Encoding",
              "routes": Array [],
            },
            Object {
              "index": 0,
              "isTransitioning": false,
              "key": "Decoding",
              "routeName": "Decoding",
              "routes": Array [],
            },
          ],
        },
      ]
    }
    labelStyle={
      Object {
        "fontFamily": "Roboto",
        "fontWeight": "normal",
      }
    }
    navigation={
      Object {
        "state": Object {
          "index": 0,
          "isDrawerOpen": false,
          "isTransitioning": false,
          "key": "Hello",
          "params": Object {
            "car_id": 123,
          },
          "routes": Array [
            Object {
              "index": 0,
              "isTransitioning": false,
              "key": "Home",
              "routeName": "Home",
              "routes": Array [
                Object {
                  "index": 0,
                  "isTransitioning": false,
                  "key": "Encoding",
                  "routeName": "Encoding",
                  "routes": Array [],
                },
                Object {
                  "index": 0,
                  "isTransitioning": false,
                  "key": "Decoding",
                  "routeName": "Decoding",
                  "routes": Array [],
                },
              ],
            },
          ],
        },
      }
    }
    screenProps={
      Object {
        "theme": Object {
          "background": "#FFF",
          "color": "#000",
          "isDark": false,
        },
      }
    }
  />
</View>
`;

exports[`DrawerNavigator: Match snapshots 2 1`] = `<View />`;
